grails-app/views/location/edit.gsp

```diff
@@ -17,10 +17,7 @@
 
 <body>
 <div class="body">
-
-    <g:if test="${flash.message}">
-        <div class="message">${flash.message}</div>
-    </g:if>
+    <warehouse:renderMessage />
```

Comment:
Using a join(".") in the controller might allow us to avoid this.

Reply:
> Can you explain the reason for this?

The reason for that was to be able to include a link inside the message. This has nothing to do with the dot.

Reply:
The reason behind creating this taglib util function for rendering messages is because when translating labels in the controller and passing them using `flash.messages` it would either escape the HTML or render the HTML as a string.

So as a solution we want to pass to the `flash.message` all of the arguments needed for rendering the translated label and let the `g.message` handle the rendering.

I implemented this `<warehouse:renderMessage />` taglib function in a way so it would be possible to render either a single message or a list of messages.
We can also both pass a normal string or as I mentioned before all of the arguments for rendering message.properties labels like **code**, **args**, **default**